source/core_sntp_serializer.c

```diff
@@ -244,10 +246,10 @@ static uint32_t readWordFromNetworkByteOrderMemory( const uint32_t * ptr )
 
     assert( ptr != NULL );
 
-    return ( uint32_t ) ( ( ( uint32_t ) *( pMemStartByte ) << 24 ) |
-                          ( 0x00FF0000U & ( ( uint32_t ) *( pMemStartByte + 1 ) << 16 ) ) |
-                          ( 0x0000FF00U & ( ( uint32_t ) *( pMemStartByte + 2 ) << 8 ) ) |
-                          ( ( uint32_t ) *( pMemStartByte + 3 ) ) );
+    return ( uint32_t ) ( ( ( uint32_t ) pMemStartByte[ 0 ] << 24 ) |
```

Comment:
I think we should add extra parenthesis around the cast and the variable.
Because here, reading it is not clear - is the variable shifted 24 bits and then casted to uint32_t or is it casted to uint32_t and then shifted 24 bits?
```suggestion
    return ( uint32_t ) ( ( ( ( uint32_t ) pMemStartByte[ 0 ] ) << 24 ) |
```

And similarly for the following lines.

Reply:
Thanks for the comment. Update in new commit.